File: swift/Sources/ShipzySDK/ComplianceClient.swift
import Foundation

// ============ Compliance Models ============

public struct ComplianceCheckResult: Codable, Sendable {
    public let compliant: Bool
    public let restrictions: [Restriction]
    public let requiredDocuments: [String]
    public let tips: [String]
}

public struct Restriction: Codable, Sendable {
    public let type: String
    public let item: String
    public let message: String
}

public struct CountryRequirements: Codable, Sendable {
    public let countryCode: String
    public let restrictions: [String]
    public let requiredDocuments: [DocumentRequirement]
    public let notes: String
}

public struct DocumentRequirement: Codable, Sendable {
    public let type: String
    public let description: String
    public let required: Bool
}

// ============ Compliance Client ============

public class ComplianceClient: HttpClient {
    public func check(_ data: [String: Any]) async throws -> ComplianceCheckResult {
        let body = try JSONSerialization.data(withJSONObject: data)
        return try await request("/api/v1/compliance/check", method: "POST", body: body)
    }

    public func countryRequirements(_ countryCode: String) async throws -> CountryRequirements {
        return try await request("/api/v1/compliance/requirements/\(countryCode)")
    }

    public func validateHsCode(_ hsCode: String) async throws -> [String: Any] {
        return try await request("/api/v1/compliance/hscode/\(hsCode)/validate")
    }

    public func prohibitedItems() async throws -> [String: Any] {
        return try await request("/api/v1/compliance/prohibited")
    }

    public func createCustoms(_ data: [String: Any]) async throws -> [String: Any] {
        let body = try JSONSerialization.data(withJSONObject: data)
        return try await request("/api/v1/compliance/customs", method: "POST", body: body)
    }

    public func getCustoms(_ id: String) async throws -> [String: Any] {
        return try await request("/api/v1/compliance/customs/\(id)")
    }
}

File: dotnet/src/Shipzy/ComplianceClient.cs
using System.Threading.Tasks;
using Shipzy.Sdk.Models;

namespace Shipzy.Sdk
{
    public class ComplianceClient : ShipzyHttpClient
    {
        public ComplianceClient(ZymeupConfig config) : base(config) { }

        public async Task<ApiResult<object>> CreateCustomsAsync(object data)
            => await RequestAsync<ApiResult<object>>("/api/v1/compliance/customs", "POST", data);

        public async Task<ApiResult<object>> GetCustomsAsync(string id)
            => await RequestAsync<ApiResult<object>>($"/api/v1/compliance/customs/{System.Uri.EscapeDataString(id)}");

        public async Task<ApiResult<object>> CheckAsync(object data)
            => await RequestAsync<ApiResult<object>>("/api/v1/compliance/check", "POST", data);

        public async Task<ApiResult<object>> GetCountryRequirementsAsync(string countryCode)
            => await RequestAsync<ApiResult<object>>($"/api/v1/compliance/requirements/{System.Uri.EscapeDataString(countryCode)}");

        public async Task<ApiResult<object>> ValidateHsCodeAsync(string hsCode)
            => await RequestAsync<ApiResult<object>>($"/api/v1/compliance/hscode/{System.Uri.EscapeDataString(hsCode)}/validate");

        public async Task<ApiResult<object>> GetProhibitedItemsAsync()
            => await RequestAsync<ApiResult<object>>("/api/v1/compliance/prohibited");
    }
}
